lib/spack/spack/package.py

```diff
@@ -1255,16 +1255,19 @@ def _if_make_target_execute(self, target):
 
         if not matches:
             tty.msg("Target '" + target + ":' not found in Makefile")
-            return
+            return False
+        return True
 
-        # Execute target
-        inspect.getmodule(self).make(target)
+    def _if_make_target_execute(self, target):
+        if self._if_make_target(target):
+            # Execute target
+            inspect.getmodule(self).make(target)
 
-    def _if_ninja_target_execute(self, target):
+    def _if_ninja_target(self, target):
```

Comment:
In general I sympathize with @davydden here. Also a minor naming comment: I think the splitted functions should be named something like `_make_has_target` and `ninja_has_target` instead of `_if_ninja_target`. That in my opinion sounds better when reading the code. For instance:
```python
if self._make_has_target('test'):
    inspect.getmodule(self).ctest()
```
versus:
```python
if self._if_make_target('test'):
    inspect.getmodule(self).ctest()
```